sam/functions/grimreaper-summarizer/app.py

```diff
@@ -0,0 +1,141 @@
+import json
+import logging
+import os
+import re
+import sys
+import csv
+import tempfile
+import datetime
+
+import boto3
+from boto3.dynamodb.conditions import Attr, Key
+
+import grimreaper_common
+
+# Logging
+log = logging.getLogger()
+log.setLevel(logging.DEBUG)
+
+# Initialize AWS services (testing is done with localstack when AWS_SAM_LOCAL env is set
+if os.getenv("AWS_SAM_LOCAL"):
+    dynamodb = boto3.resource(
+        'dynamodb', endpoint_url="http://host.docker.internal:8000/")
+else:
+    dynamodb = boto3.resource('dynamodb')
+    s3 = boto3.client('s3')
+
+    # AWS X-Ray
+    # from aws_xray_sdk.core import xray_recorder
+    from aws_xray_sdk.core import patch_all
+    patch_all()
+
+# Initialize variables
+stage = os.environ["STAGE"]
+table_userprocessing = dynamodb.Table(os.environ['USER_PROCESSING_TABLE'])
+s3_bucket = os.environ["S3_BUCKET_NAME"]
+s3_obj_key = "user_report/{:%Y-%m-%d-%H-%M-%S}.csv".format(datetime.datetime.now())
+
+
+def handler(event, context):
+    log.debug("Received event {}".format(json.dumps(event)))
+
+    all_users = get_users(event.get('job_uuid'))
+    active_users, inactive_users = bucket_users(all_users)
+
+    user_summary(event.get('job_uuid'), inactive_users, 'inactive')
+    user_summary(event.get('job_uuid'), active_users, 'active')
+
+    active_user_report_link = generate_url(s3_obj_key, active_users, 'active')
+
+    return {'job_uuid': event.get('job_uuid'), 'active_user_report_link': active_user_report_link}
+
+
+def user_summary(job_uuid, user_list, category):
+
+    table_userprocessing.put_item(
+        Item={
+            "pk": job_uuid,
+            "sk": 'COUNT_' + str(category).upper(),
+            "count": user_list.get(str(category).lower()).get('count')
+        }
+    )
+
+
+def get_users(job_uuid):
+
+    all_users = list()
+
+    response = table_userprocessing.query(
+        KeyConditionExpression=Key('pk').eq(
+            job_uuid) & Key('sk').begins_with('USER_')
+    )
+
+   # Paginate through results 
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [indentation is not a multiple of four (comment)](https://app.codacy.com/manual/alanwill/slack-grim-reaper/pullRequest?prid=4805070)